packages/medusa/src/loaders/defaults.ts

```diff
@@ -185,3 +145,113 @@ export default async ({
     }
   })
 }
+
+async function registerPaymentProvider({
+  manager,
+  container,
+  logger,
+}: {
+  container: AwilixContainer
+  manager: EntityManager
+  logger: Logger
+}): Promise<void> {
+  const payProviders =
+    silentResolution<
+      (
+        | typeof BasePaymentService
+        | AbstractPaymentService
+        | AbstractPaymentProcessor
+      )[]
+    >(container, "paymentProviders", logger) || []
+
+  const payIds: string[] = []
+  await Promise.all(
+    payProviders.map((paymentProvider) => {
+      payIds.push(paymentProvider.getIdentifier())
+
+      if (paymentProvider instanceof AbstractPaymentProcessor) {
+        return paymentProvider.init()
+      }
+
+      return
+    })
+  )
+
+  const pProviderService = container.resolve<PaymentProviderService>(
+    "paymentProviderService"
+  )
+  await pProviderService
+    .withTransaction(manager)
+    .registerInstalledProviders(payIds)
+}
```

Comment:
thought: would it make sense to completely separate PaymentProcessors from the legacy payment provider service. 

My thinking is that we would then be able to avoid all the acrobatics that are required to make it backward compatible? 

I wouldn't mind for example if we hide the "legacy providers" somewhere in the admin dashboard. And then make people aware that they will soon be deprecated.

Reply:
Also, I mentioned to oli the other day, that at that time we might want to review the payment-provider service API. Maybe splitting everything related to the payment session into his own service, exposing the API to manage the payments (locally), and re aligned the method naming that communicates with the plugin. Something to discuss

Reply:
Just bringing this discussion to life. I think it would make sense to separate payment processor registration into its own method `registerPaymentProcessors` to avoid mixing these two concepts together. The upside is not huge, but it separates the concerns and allows for "easier" clean up later. 

Also, I don't see why we shouldn't.

Reply:
oups sorry, I might have miss understood the first time, yeah completely